### DISCUSSION TITLE
Timestep manipulations

### DISCUSSION THREAD

**Discussion by Q-R-B** [DHI/mikeio#566] [Q&A] [answered]
Hi,

I have a dataset with one item and two timesteps.
![image](https://user-images.githubusercontent.com/103583383/234305243-6c54ffcf-18e5-4c35-9cc7-7c48548421be.png)

And i have an array (of same dimention).
![image](https://user-images.githubusercontent.com/103583383/234305409-3d910746-bc23-4124-9c0d-f7d4a30e00e8.png)

my question is: In the dataset, can I replace the item data for the second timestep with the data from the array?

hope it's clear (my end goal is to create at dfsu file with with two timesteps, one with initial water levels and the other one with maximum water levels)

thanks for the help

**Comment by ecomodeller** [comment]
I guess you can, but the question is why? Isn't it better to create a new file with the max wl?

**Reply by Q-R-B** [reply]
Because to visualise a cross section in qgis with the crayfish plugin, you can't select two different items (ground level and maximum water level for example). The only option is to select different time steps within the same item.

Is there any fonction to do this?

**Reply by ecomodeller** [reply, marked as answer]
I am not sure I still get the problem, but you can certainly set or change the values of a datarray, the values attributes is a regular numpy array. Also for a specific timestep.
```python
ds[0].values[1,:] = dsmax[0].values # set values of the second timestep to the aggregated values from other dataset
ds.to_dfs("newfile.dfsu")
```

Give it a try.

**Reply by Q-R-B** [reply]
Thank you, works well :) allows me to visualise ground level and max WL for different scenarios by selecting different timesteps. It's a nice workaround for now.
![Skärmbild 2023-06-19 150331](https://github.com/DHI/mikeio/assets/103583383/5ea3fda9-d16a-4552-b674-008e764c9a47)
